frigate/ffmpeg_presets.py

```diff
@@ -91,8 +91,8 @@ def get_selected_gpu(self) -> str:
     FFMPEG_HWACCEL_NVIDIA: "-r {0} -vf fps={0},scale_cuda=w={1}:h={2}:format=nv12,hwdownload,format=nv12,format=yuv420p",
     "preset-jetson-h264": "-r {0}",  # scaled in decoder
     "preset-jetson-h265": "-r {0}",  # scaled in decoder
-    "preset-rk-h264": "-r {0} -vf fps={0},scale={1}:{2}",
-    "preset-rk-h265": "-r {0} -vf fps={0},scale={1}:{2}",
+    "preset-rk-h264": "-r {0} -vf scale_rkrga=w={1}:h={2}:format=nv12,hwdownload,format=nv12,format=yuv420p",
```

Comment:
Is there any reason not to convert directly to `yuv420p` on hardware? nv12->yuv420p is not free on the cpu. Same for the `scale_cuda`. If you use `yuv420p` then the decoder cannot enable `afbc` because rga2 does not support it.

In addition, for rockchip, using `hwmap`(reference) instead of `hwdownload`(duplicate) can avoid extra memcpy.

Reply:
we found that doing the conversion on hardware caused contrast to be lost because the full color range was not returned by the hardware https://github.com/blakeblackshear/frigate/pull/8604 https://github.com/blakeblackshear/frigate/discussions/8645#discussion-5859536

Reply:
AFAIK the `scale_vaapi` filter doesn’t support fully-planar formats, such as `yuv420p`. So I’m surprised that vaapi can even work properly before #8604.

And the difference between `nv12` and `yuv420p` is only in the memory layout. If you really want to request the output of a specific color range, please refer to [this](https://github.com/FFmpeg/FFmpeg/blob/e43615fc2ab27d562ed7e087803f4a364a7d1175/libavfilter/vf_scale_vaapi.c#L241) option.

Reply:
yeah, I tried using that option while trying to fix but unfortunately it seemed to be ignored. #8604 was the only fix, perhaps due to limited support like you said.

Currently we require yuv420p, supporting nv12 as well is an improvement that will be looked at in the future

Reply:
Hey @nyanmisaka,
Thank you for reviewing. Will these contrast issues also appear on rockchip devices?

P.S. You were interested in building a static mpp lib. I have a [fork here](https://github.com/MarcA711/mpp) that does it by applying `-DBUILD_SHARED_LIBS=OFF`, for example: `cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/data/build -DBUILD_TEST=OFF -DBUILD_SHARED_LIBS=OFF ..`. However, I just call a bash script that combines all static libs using gnu `ar`. So, this is not platform independent. The best way is probably to rework all `CMakeLists.txt` files.

Reply:
There shouldn't be any issues as it is consistent with other commands, when we support other formats in the future then all the commands can be updated at the same time

Reply:
> Hey @nyanmisaka, Thank you for reviewing. Will these contrast issues also appear on rockchip devices?
> 
> P.S. You were interested in building a static mpp lib. I have a [fork here](https://github.com/MarcA711/mpp) that does it by applying `-DBUILD_SHARED_LIBS=OFF`, for example: `cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/data/build -DBUILD_TEST=OFF -DBUILD_SHARED_LIBS=OFF ..`. However, I just call a bash script that combines all static libs using gnu `ar`. So, this is not platform independent. The best way is probably to rework all `CMakeLists.txt` files.

Jellyfin's ffmpeg requires libvpx, but as you [said here](https://github.com/MarcA711/Rockchip-FFmpeg-Builds/releases), mpp and libvpx conflict. So I still need to wait for the rockchip guys to replace the corresponding function names.